bin/agitator

```diff
@@ -40,60 +115,84 @@ function start_agitator() {
   else
     echo >&2 "The agitator requires pssh/parallel-ssh to be installed. Aborting."; exit 1;
   fi
+  ## read configuration into env variables
+  read_cluster_conf
 
   mkdir -p "${at_home}/logs"
   log_base="${at_home}/logs/$(date +%Y%m%d%H%M%S)_$(hostname)"
-  libexec="${at_home}/libexec"
-  master_log="${log_base}_master-agitator"
-  tserver_log="${log_base}_tserver-agitator"
-  datanode_log="${log_base}_datanode-agitator"
-  master_cmd="nohup ${libexec}/master-agitator.pl $AGTR_MASTER_KILL_SLEEP_TIME $AGTR_MASTER_RESTART_SLEEP_TIME"
-  tserver_cmd="nohup ${libexec}/tserver-agitator.pl $AGTR_TSERVER_KILL_SLEEP_TIME $AGTR_TSERVER_RESTART_SLEEP_TIME $AGTR_TSERVER_MIN_KILL $AGTR_TSERVER_MAX_KILL"
-  datanode_cmd="nohup ${libexec}/datanode-agitator.pl $AGTR_DATANODE_KILL_SLEEP_TIME $AGTR_DATANODE_RESTART_SLEEP_TIME $HADOOP_HOME $AGTR_DATANODE_MIN_KILL $AGTR_DATANODE_MAX_KILL"
+  manager_log="${log_base}_manager-agitator.log"
+  tserver_log="${log_base}_tserver-agitator.log"
+  datanode_log="${log_base}_datanode-agitator.log"
+  datanode_kill_cmd="pkill -9 -f '[p]roc_datanode'"
+  datanode_start_cmd="$HADOOP_HOME/bin/hdfs --daemon start datanode"
+  manager_kill_cmd="pkill -f '[ ]org[.]apache[.]accumulo[.]start[.]Main manager'"
+  manager_start_cmd="$ACCUMULO_HOME/bin/accumulo-service manager start"
+  tserver_kill_cmd="pkill -f '[ ]org[.]apache[.]accumulo[.]start[.]Main tserver'"
+  tserver_start_cmd="$ACCUMULO_HOME/bin/accumulo-service tserver start"
+
   [[ -n $AGITATOR_USER ]] || AGITATOR_USER=$(whoami)
 
-  if [[ $AGITATOR_USER == "$AGTR_ACCUMULO_USER" ]]; then
-    echo "Running master-agitator and tserver-agitator as $AGITATOR_USER"
-    $master_cmd > "${master_log}.out" 2> "${master_log}.err" &
-    $tserver_cmd > "${tserver_log}.out" 2> "${tserver_log}.err" &
-  else
-    echo "Running master-agitator and tserver-agitator as $AGTR_ACCUMULO_USER using sudo."
-    sudo --preserve-env=PSSH -u "$AGTR_ACCUMULO_USER" $master_cmd > "${master_log}.out" 2> "${master_log}.err" &
-    sudo -u "$AGTR_ACCUMULO_USER" $tserver_cmd > "${tserver_log}.out" 2> "${tserver_log}.err" &
+  if [[ $AGITATOR_USER != "$AGTR_ACCUMULO_USER" ]]; then
+    sudo -i -u "$AGTR_ACCUMULO_USER"
   fi
-  if [[ $AGITATOR_USER == "$AGTR_HDFS_USER" ]]; then
-    echo "Running datanode-agitator as $AGITATOR_USER"
-    $datanode_cmd > "${datanode_log}.out" 2> "${datanode_log}.err" &
-  else
-    echo "Running datanode-agitator as $AGTR_HDFS_USER using sudo."
-    sudo -u "$AGTR_HDFS_USER" $datanode_cmd > "${datanode_log}.out" 2> "${datanode_log}.err" &
+  echo "Starting manager and tserver agitation as $(whoami)"
+  start_app_agitator manager "$AGTR_MASTER_KILL_SLEEP_TIME" "$AGTR_MASTER_RESTART_SLEEP_TIME" 1 1 "$manager_start_cmd" "$manager_kill_cmd" > "$manager_log" 2>&1 &
```

Comment:
Are these variables, with "master" in them, `"$AGTR_MASTER_KILL_SLEEP_TIME"` and `"$AGTR_MASTER_RESTART_SLEEP_TIME"` still valid?

Reply:
Yeah I haven't changed them in `env.sh` because I thought it might be a bigger change to migrate away from master. I was going to do that in a separate change but I can here since I am modifying both files.

Reply:
> I was going to do that in a separate change but I can here since I am modifying both files.

Either way works. I just wanted to double check that it wasn't being overlooked.